### PR TITLE
chore: [IOPID-1509], [IOPID-1484] Integrate new DS on email insert/validation in onboarding flow and integrate new MP events

### PR DESCRIPTION
**Pull request by Ladirico**
## Short description
Integrate new DS on email insert/validation flow, codebase cleaning and integration of MP events

## List of changes proposed in this pull request
- Integrate new DS on email insert/validation flow
- Fix FAQ visualization
- Change flow 
- Add ProgressLoader component in validation email screen
- Change logic of polling in validation email screen (Add Saga)
- Change copy
- Add mixpanel events 
- Refactor code 
   - Transform validation email modal into screen
   - Delete unused screens
   - Change some routes name

## Test

> [!TIP]
> [Run e2e tests](https://github.com/pagopa/io-app/actions/workflows/test-e2e.yml)

<p>

| ✅ NEW FLOW - EMAIL NOT TAKEN ✅ | ✅ NEW FLOW - EMAIL ALREADY TAKEN ✅  |  ✅ NEW FLOW - WITHOUT EMAIL ✅|
| - | - | - |
| <video src="https://github.com/pagopa/io-app/assets/83651704/08bcb6e7-39a3-4315-90d0-eea2bd59f3d6" /> | <video src="https://github.com/pagopa/io-app/assets/83651704/448a0dcf-d54b-41f0-a077-12279bb9cbc2" /> | <video src="https://github.com/pagopa/io-app/assets/83651704/f77d0162-6fd2-491c-8e6c-9b4e305cf3dd" /> |
</p>

### Accessibility Test
<p>

| 🍏 A11Y iOS 🍏 | 🤖 A11Y Android 🤖 |
| - | - |
| <video src="https://github.com/pagopa/io-app/assets/83651704/0ab1de92-2d55-4a5b-91e9-d17bf49fdace" /> | <video src="https://github.com/pagopa/io-app/assets/83651704/d6bd599b-0819-4f62-b9e9-c7824c7094c3" /> |
</p>

## How to test
Run the application and test:
1.  _firstOnboarding_ flow using `io-dev-api-server`:  `is_email_validated": false`, and`"is_email_already_taken": true` (+ close the application during the validation)
2. _firstOnboarding_ flow using `io-dev-api-server`:  `is_email_validated": false`, and`"is_email_already_taken": false` (+ close the application during the validation)
6. Test too the email change in profile flow to check that changes on screens have not led to regressions

### CODE REVIEW

**Review by francescopersico — CHANGES_REQUESTED** [review]

File: ts/screens/profile/EmailValidationScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+    // if the verification email was never sent, we send it
+    if (sendEmailAtFirstRender) {
+      sendEmailValidation();
+    }
+  });
+
+  const handleContinue = () => {
+    if (isEmailValidated) {
+      trackEmailValidationSuccessConfirmed(flow);
+      navigation.goBack();
```
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/screens/profile/EmailValidationScreen.tsx]: thanks for the suggestion, it was not needed, navigation was already handled by the dispatch action

File: ts/screens/profile/EmailValidationScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+  };
+
+  const navigateToInsertEmail = () => {
+    dispatchAcknowledgeOnEmailValidation(O.none);
+    navigation.goBack();
+  };
+
+  type CountdownProps = {
+    visible: boolean;
+    timerElapsed?: () => void;
```
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/screens/profile/EmailValidationScreen.tsx]: done! ✅

File: ts/screens/profile/EmailValidationScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+  };
+
+  const Countdown = (props: CountdownProps) => {
```
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/screens/profile/EmailValidationScreen.tsx]: done, thanks for the suggestion! ✅

**Comment by Ladirico** [comment]
@mariateresaventura please check the copy! 😁

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: ts/store/sagas/emailValidationPollingSaga.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  //   yield* put(emailValidationPollingStart());
+
+  while (profilePollingIsRunning) {
+    yield* call(startTimer, GET_PROFILE_POLLING_INTERVAL);
+
+    yield* put(profileLoadRequest());
+    const isEmailValidationSelector = yield* select(emailValidationSelector);
+    profilePollingIsRunning =
+      isEmailValidationSelector.isEmailValidationPollingRunning;
+    yield* take(profileLoadSuccess);
```
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/store/sagas/emailValidationPollingSaga.ts:26]: Sorry, I don't understand the reason for this change, could you explain?

File: ts/store/sagas/emailValidationPollingSaga.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  profileLoadSuccess
+} from "../actions/profile";
+import { emailValidationSelector } from "../reducers/emailValidation";
+
+const GET_PROFILE_POLLING_INTERVAL = 5000 as Millisecond;
+
+function* emailValidationPollingLoop() {
+  // eslint-disable-next-line functional/no-let
+  let profilePollingIsRunning = true;
+  //   yield* put(emailValidationPollingStart());
```
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/store/sagas/emailValidationPollingSaga.ts:17]: I'm sorry, I don't understand, you are suggesting an as-is solution 😅

File: ts/navigation/OnboardingNavigator.tsx [hunk 6]
```diff
@@ -15,97 +12,83 @@ import FingerprintScreen from "../screens/onboarding/biometric&securityChecks/Fi
...
+import EmailValidationSendEmailScreen from "../screens/profile/EmailValidationSendEmailScreen";
 import { OnboardingParamsList } from "./params/OnboardingParamsList";
 import ROUTES from "./routes";
 
 const Stack = createStackNavigator<OnboardingParamsList>();
 /**
  * The onboarding related stack of screens of the application.
  */
-const OnboardingNavigator = () => {
-  const isEmailUniquenessValidationEnabled = useIOSelector(
```
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/navigation/OnboardingNavigator.tsx:28]: done!

File: ts/navigation/ProfileNavigator.tsx [hunk 7]
```diff
@@ -44,172 +41,147 @@ const Stack = createStackNavigator<ProfileParamsList>();
 /**
  * A navigator for all the screens of the Profile section
  */
-const ProfileStackNavigator = () => {
-  const isEmailUniquenessValidationEnabled = useIOSelector(
-    isEmailUniquenessValidationEnabledSelector
-  );
```
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/navigation/ProfileNavigator.tsx:50]: done!

File: ts/navigation/ProfileNavigator.tsx [hunk 8]
```diff
@@ -44,172 +41,147 @@ const Stack = createStackNavigator<ProfileParamsList>();
...
-          name={ROUTES.PROFILE_PREFERENCES_NOTIFICATIONS}
-          component={NotificationsPreferencesScreen}
-        />
-      )}
-    </Stack.Navigator>
-  );
-};
+    <Stack.Screen
+      name={ROUTES.INSERT_EMAIL_SCREEN}
+      component={CduEmailInsertScreen}
```
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/navigation/ProfileNavigator.tsx]: done!

File: ts/navigation/OnboardingNavigator.tsx [hunk 9]
```diff
@@ -15,97 +12,83 @@ import FingerprintScreen from "../screens/onboarding/biometric&securityChecks/Fi
...
+      component={MissingDevicePinScreen}
+    />
+    <Stack.Screen
+      options={{ headerShown: false }}
+      name={ROUTES.ONBOARDING_MISSING_DEVICE_BIOMETRIC}
+      component={MissingDeviceBiometricScreen}
+    />
+    <Stack.Screen
+      name={ROUTES.ONBOARDING_INSERT_EMAIL_SCREEN}
+      component={CduEmailInsertScreen}
```
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/navigation/OnboardingNavigator.tsx]: done!

File: ts/sagas/startup/checkAcknowledgedEmailSaga.ts [hunk 10]
```diff
@@ -48,21 +48,13 @@ export function* checkAcknowledgedEmailSaga(
     }
   } else {
     // the profile has no email address, user must insert it
-    // EmailInsertScreen knows if the user comes from onboarding or not
+    // cduEmailScreen knows if the user comes from onboarding or not
```
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/sagas/startup/checkAcknowledgedEmailSaga.ts]: done!

File: ts/screens/profile/ProfileDataScreen.tsx [hunk 11]
```diff
@@ -28,7 +28,7 @@ const contextualHelpMarkdown: ContextualHelpPropsMarkdown = {
 const ProfileDataScreen: React.FC<Props> = ({
   profileEmail,
   isEmailValidated,
-  isEmailUniquenessValidationEnabled,
+  // isEmailUniquenessValidationEnabled,
```
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/screens/profile/ProfileDataScreen.tsx]: done!

File: ts/screens/profile/__test__/ProfileDataScreen.test.tsx [hunk 12]
```diff
@@ -69,14 +69,14 @@ describe("Test ProfileDataScreen", () => {
...
-    expect(component).not.toBeNull();
-    const listItemComponent = component.getByTestId("insert-or-edit-email");
-    expect(listItemComponent).not.toBeNull();
-    fireEvent.press(listItemComponent);
-  });
+  //   expect(component).not.toBeNull();
+  //   const listItemComponent = component.getByTestId("insert-or-edit-email");
+  //   expect(listItemComponent).not.toBeNull();
+  //   fireEvent.press(listItemComponent);
+  // });
```
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/screens/profile/__test__/ProfileDataScreen.test.tsx]: I think I commented out those lines of code by mistake, I uncommented them!

File: ts/screens/profile/__test__/ProfileDataScreen.test.tsx [hunk 13]
```diff
@@ -69,14 +69,14 @@ describe("Test ProfileDataScreen", () => {
       ).not.toBeNull();
     }
   });
-  it("when press ListItemComponent insert or edit email, if user has email should navigate to EmailReadScreen else should navigate to EmailInsertScreen", () => {
-    const { component } = renderComponent();
+  // it("when press ListItemComponent insert or edit email, if user has email should navigate to EmailReadScreen else should navigate to EmailInsertScreen", () => {
+  //   const { component } = renderComponent();
```
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/screens/profile/__test__/ProfileDataScreen.test.tsx]: I think I commented out those lines of code by mistake, I uncommented them!

File: ts/screens/profile/EmailValidationSendEmailScreen.tsx [hunk 14]
```diff
@@ -0,0 +1,344 @@
...
+  const { isOnboarding, sendEmailAtFirstRender } = props;
+  const headerHeight = useHeaderHeight();
+  const dispatch = useIODispatch();
+  const navigation = useIONavigation();
+  const optionEmail = useIOSelector(
+    profileEmailSelector,
+    (l, r) =>
+      (O.isNone(l) && O.isNone(r)) ||
+      (O.isSome(l) && O.isSome(r) && l.value === r.value)
+  );
```
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/screens/profile/EmailValidationSendEmailScreen.tsx:91]: done! Thanks for the suggestion!

File: ts/store/sagas/emailValidationPollingSaga.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (shadowsheep1) [reply ts/store/sagas/emailValidationPollingSaga.ts]: I suggest to remove the comment ;)

File: ts/screens/profile/EmailValidationSendEmailScreen.tsx [hunk 15]
```diff
@@ -0,0 +1,344 @@
...
+};
+const contextualHelpMarkdown: ContextualHelpPropsMarkdown = {
+  title: "email.validate.title",
+  body: "email.validate.help"
+};
+const EmailValidationSendEmailScreen = () => {
+  const props =
+    useRoute<
+      Route<
+        "ONBOARDING_READ_EMAIL_SCREEN" | "PROFILE_EMAIL_INSERT_SCREEN",
```
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/screens/profile/EmailValidationSendEmailScreen.tsx:79]: done!

File: ts/store/sagas/emailValidationPollingSaga.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (shadowsheep1) [reply ts/store/sagas/emailValidationPollingSaga.ts:25]: Yep! You are dispatching `profileLoadRequest()`, this trigger an API call that reload the citizen profile. Before checking the validity of the email, you should wait for the API call to end: the take of `profileLoadSuccess` (API call succesfully ended) should be before checking the validity of the email. What do you think?

File: ts/store/sagas/emailValidationPollingSaga.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/store/sagas/emailValidationPollingSaga.ts]: done!

File: ts/store/sagas/emailValidationPollingSaga.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/store/sagas/emailValidationPollingSaga.ts:25]: oh yes! thanks!

[the remark this answers is not in the document]
Comment (Ladirico) [reply ts/store/sagas/emailValidationPollingSaga.ts]: done!

**Review by shadowsheep1 — COMMENTED** [review]
I've found a critical path:
- User already logged in.
- User with invalid email.
- User go straight ahead to email validation.
- After a successful validation is stuck in a wrong navigation stack.

Are you able to reproduce it or am I missing something?

<details><summary>Details</summary>
<p>

| Wrong Way ❌ |
| - | 
| <video src="https://github.com/pagopa/io-app/assets/16268789/c8487f28-b3cc-4bae-92e8-76febedb02e6" /> |

</p>
</details>

**Comment by shadowsheep1** [comment]
We close it due to a wrong rebase.

**Pull request closed without merging**